simple_repository/components/yanking.py

```diff
@@ -124,35 +121,51 @@ async def yanked_files(
         yanked_files = {}
         value = self._yank_config.get(project_page.name)
         if value:
-            pattern, reason = value
-            yanked_files = {
-                file.filename: reason
-                for file in project_page.files
-                if fnmatch.fnmatch(file.filename, pattern)
-            }
-
+            untouched_files = list(project_page.files)
+            for pattern, reason in value.items():
+                for file in untouched_files.copy():
```

Comment:
There will be many more files than copies, so I would suggest iterating in the other order.

Reply:
I applied this directly.